presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -133,6 +140,18 @@
     Block getRegion(int positionOffset, int length);
 
     /**
+     * Returns a block starting at the specified position and extends for the
+     * specified length.  The specified region must be entirely contained
+     * within this block.
+     *
+     * The region returned must be a compact representation of the original block, unless their internal
+     * representation will be exactly the same. This method is useful for
+     * operators that hold on to a range of values without holding on to the
+     * entire block.
+     */
+    Block getCompactedRegion(int position, int length);
```

Comment:
We needed this early, so we merged your changes for the existing blocks and named it `copyRegion`